In [1]:
import pandas as pd
import numpy as np
from statsmodels.stats.proportion import proportion_confint

file_path = "/kaggle/input/global-air-pollution-dataset/global air pollution dataset.csv"
data = pd.read_csv(file_path)

data.replace([np.inf, -np.inf], np.nan, inplace=True)
data = data.dropna()

In [2]:
def estimate_sample_size(p, E, Z=1.96):
    n = (Z**2 * p * (1 - p)) / E**2
    return np.ceil(n)

# розрахункова частка p (максимальна варіабельність), похибка E
p = 0.5
E = 0.05

sample_size = estimate_sample_size(p, E)
sample_size

385.0

Як бачимо обсяг вибірки, необхідний для досягнення похибки 0,05 з рівнем довіри 95% і розрахунковою часткою 0,5, становить 385.

Далі оцінимо частку кожної категорії AQI (поле `AQI Category`) і розрахуємо довірчий інтервал для цієї частки за допомогою біноміального розподілу.

In [3]:
aqi_categories = data['AQI Category'].value_counts()
total_count = len(data)

results = {}

for category, count in aqi_categories.items():
    proportion = count / total_count
    confidence_interval = proportion_confint(count, total_count, alpha=0.05, method='binom_test')
    results[category] = (proportion, confidence_interval)

for category, (proportion, confidence_interval) in results.items():
    print(f"AQI Category: {category}")
    print(f"Proportion: {proportion:.4f}")
    print(f"95% Confidence Interval: {confidence_interval}\n")

AQI Category: Good
Proportion: 0.4206
95% Confidence Interval: (0.4141947133957241, 0.42697977269815246)

AQI Category: Moderate
Proportion: 0.3945
95% Confidence Interval: (0.38819051771529467, 0.40082360759007496)

AQI Category: Unhealthy
Proportion: 0.0962
95% Confidence Interval: (0.0923977130494913, 0.1000384197363583)

AQI Category: Unhealthy for Sensitive Groups
Proportion: 0.0681
95% Confidence Interval: (0.06487407773769029, 0.07140759741688646)

AQI Category: Very Unhealthy
Proportion: 0.0124
95% Confidence Interval: (0.01104170600664649, 0.01392917345478256)

AQI Category: Hazardous
Proportion: 0.0083
95% Confidence Interval: (0.007177841765914954, 0.009544550334734081)

